mixer/pkg/perf/controller.go

```diff
@@ -92,18 +99,47 @@ func (c *Controller) initializeClients(address string, setup *Setup) error {
 	return err
 }
 
-func (c *Controller) runClients(iterations int) error {
-	var err error
+func (c *Controller) runClients(iterations int, timeout time.Duration) error {
+	if len(c.clients) == 0 {
+		return nil
+	}
+
+	if timeout == 0 {
+		timeout = defaultTimeout
+	}
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+
+	resErr := []error{}
+	resCount := 0
+	errCh := make(chan error, len(c.clients))
+
 	for _, conn := range c.clients {
-		// TODO: This needs to be an async call when we have more than 1 client.
-		e := conn.Call("ClientServer.Run", iterations, nil)
-		if e != nil && err == nil {
-			// Capture the first error
-			err = e
+		// Make calls asynchronously.
+		go func() { errCh <- conn.Call("ClientServer.Run", iterations, nil) }()
+	}
+
+	for {
+		if resCount >= len(c.clients) {
+			// All of the calls returned
+			break
+		}
+		select {
+		case e := <-errCh:
+			resCount++
+			if e != nil {
+				resErr = append(resErr, e)
+			}
+		case <-timer.C:
+			return fmt.Errorf("timeout waiting for call response")
 		}
 	}
 
-	return err
+	if len(resErr) > 0 {
+		return errorsutil.AggregateError(resErr)
```

Comment:
Most of the Istio code is using multierror.Append to accumulate errors.  We should standardize on that.